clang/docs/ReleaseNotes.rst

```diff
@@ -46,6 +46,9 @@ C/C++ Language Potentially Breaking Changes
 
 C++ Specific Potentially Breaking Changes
 -----------------------------------------
+- The `ClassScopeFunctionSpecializationDecl` AST node has been removed. Dependent class scope
+explicit function template specializations now use `DependentFunctionTemplateSpecializationInfo`
+to store candidate primary templates and explicit template arguments.
```

Comment:
```suggestion
to store candidate primary templates and explicit template arguments. This should not impact users
of Clang as a compiler, but it may break assumptions in Clang-based tools iterating over the AST.
```